.evergreen/.evg.yml

```diff
@@ -288,6 +288,176 @@ functions:
+        script: |
+          # DO NOT ECHO WITH XTRACE (which PREPARE_SHELL does)
+
+          export DRIVERS_TOOLS="$DRIVERS_TOOLS"
+
+          alias urlencode='python -c "import sys, urllib as ul; print ul.quote_plus(sys.argv[1])"'
+
+          USER=$(jq -r '.AccessKeyId' ${DRIVERS_TOOLS}/.evergreen/auth_aws/creds.json)
+          USER=$(urlencode $USER)
+
+          PASS=$(jq -r '.SecretAccessKey' ${DRIVERS_TOOLS}/.evergreen/auth_aws/creds.json)
+          PASS=$(urlencode $PASS)
+
+          SESSION_TOKEN=$(jq -r '.SessionToken' ${DRIVERS_TOOLS}/.evergreen/auth_aws/creds.json)
+          SESSION_TOKEN=$(urlencode $SESSION_TOKEN)
+
+          JDK=${JDK} SESSION_TOKEN=$SESSION_TOKEN DRIVERS_TOOLS=${DRIVERS_TOOLS} MONGODB_URI="mongodb://$USER:$PASS@localhost" .evergreen/run-mongodb-aws-test.sh
+
+  "configure aws EC2 credentials":
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: "src"
+        script: |
+          ${PREPARE_SHELL}
+          cd ${DRIVERS_TOOLS}/.evergreen/auth_aws
+          ${MONGODB_BINARIES}/mongo --verbose aws_e2e_ec2.js
+          cd -
+
+  "run aws auth test with aws EC2 credentials":
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: "src"
+        silent: true
+        script: |
+          # DO NOT ECHO WITH XTRACE (which PREPARE_SHELL does)
+          JDK=${JDK} .evergreen/run-mongodb-aws-test.sh
+
+  "run aws auth test with aws credentials as environment variables":
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: "src"
+        silent: true
+        script: |
+          # DO NOT ECHO WITH XTRACE (which PREPARE_SHELL does)
+
+          alias urlencode='python -c "import sys, urllib as ul; print ul.quote_plus(sys.argv[1])"'
+          export AWS_ACCESS_KEY_ID=$(urlencode ${iam_auth_ecs_account})
+          export AWS_SECRET_ACCESS_KEY=$(urlencode ${iam_auth_ecs_secret_access_key})
+
+          JDK=${JDK} .evergreen/run-mongodb-aws-test.sh
+
+  "run aws auth test with aws credentials and session token as environment variables":
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: "src"
+        silent: true
+        script: |
+          # DO NOT ECHO WITH XTRACE (which PREPARE_SHELL does)
+          export DRIVERS_TOOLS="$DRIVERS_TOOLS"
+
+          alias urlencode='python -c "import sys, urllib as ul; print ul.quote_plus(sys.argv[1])"'
+
+          USER=$(jq -r '.AccessKeyId' ${DRIVERS_TOOLS}/.evergreen/auth_aws/creds.json)
+          export AWS_ACCESS_KEY_ID=$(urlencode $USER)
+
+          PASS=$(jq -r '.SecretAccessKey' ${DRIVERS_TOOLS}/.evergreen/auth_aws/creds.json)
+          export AWS_SECRET_ACCESS_KEY=$(urlencode $PASS)
+
+          SESSION_TOKEN=$(jq -r '.SessionToken' ${DRIVERS_TOOLS}/.evergreen/auth_aws/creds.json)
+          export AWS_SESSION_TOKEN=$(urlencode $SESSION_TOKEN)
+
+          JDK=${JDK} .evergreen/run-mongodb-aws-test.sh
+
+  "run aws ECS auth test":
```

Comment:
@ShaneHarvey Here's the ECS auth test. But yes, it still needs some work so you'll see the function name comment out  on line 636.

Reply:
This is done now.